### PR TITLE
feat(generator): Generate mix.exs and README

### PR DESCRIPTION
**Pull request by dazuma**
The generator now generates `mix.exs` and `README.md`. This is to solve two issues:
1. All libraries need to update their `google_gax` dependency from `~> 0.1.0` to `~> 0.2.0` in order to support sending the library version in the `x-goog-api-client` header.
2. The readme needs to show an up to date version requirement in the installation instructions. Currently, most if not all readmes are showing `~> 0.0.1` which of course no longer works for any library.

Some notes on the generation of these files:
* The mixfile namespace no longer includes the version. (e.g. it is `GoogleApis.VideoIntelligence.Mixfile` rather than `GoogleApis.VideoIntelligence.V1beta1.Mixfile`)
* Generated mixfiles now include a requirement of `google_gax ~> 0.2.0`. This means this PR cannot go into effect until gax 0.2.0 is released. (See #2149 )
* Updated Elixir requirement from `~> 1.4` to `~> 1.6`.
* The description is set to `Client library for <%= api_title %> from Google.` where `api_title` comes from the `title` field in the discovery doc
* The product homepage link is set to the `documentationLink` field in the discovery doc.
* The version requirement in the example install instructions is semver-compatible. For example, if the library version is `0.4.2`, the install instructions say to use `~> 0.4`.

### CODE REVIEW

File: template/elixir/mix.exs.eex [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    """
+  end
+
+  defp package() do
+    [
+      files: ["lib", "mix.exs", "README*", "LICENSE"],
+      maintainers: ["Jeff Ching", "Daniel Azuma"],
+      licenses: ["Apache 2.0"],
+      links: %{
+        "GitHub" => "https://github.com/GoogleCloudPlatform/elixir-google-api/tree/master/clients/<%= library_name %>",
```
Comment (chingor13) [inline template/elixir/mix.exs.eex]:
```suggestion
        "GitHub" => "https://github.com/googleapis/elixir-google-api/tree/master/clients/<%= library_name %>",
```

File: template/elixir/mix.exs.eex [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    [
+      app: :google_api_<%= library_name %>,
+      version: @version,
+      elixir: "~> 1.6",
+      build_embedded: Mix.env == :prod,
+      start_permanent: Mix.env == :prod,
+      description: description(),
+      package: package(),
+      deps: deps(),
+      source_url: "https://github.com/GoogleCloudPlatform/elixir-google-api/tree/master/clients/<%= library_name %>"
```
Comment (chingor13) [inline template/elixir/mix.exs.eex]:
```suggestion
      source_url: "https://github.com/googleapis/elixir-google-api/tree/master/clients/<%= library_name %>"
```

**Comment by dazuma** [comment]
In `api.ex.eex` I think we're going to need to get the library version by calling `Application.spec/2` at runtime, instead of `Mix.Project.config/0` at compile time. The latter doesn't work; it gets the end-user's project instead of the client library project.

**Comment by dazuma** [comment]
Turns out, after testing, that `Mix.Project.config/0` works after all. Dependencies are compiled under their own mix project.

**Pull request merged**